services/src/test/scala/cromwell/pipeline/service/UserServiceTestImpl.scala

```diff
@@ -0,0 +1,53 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dto.user.{ PasswordUpdateRequest, UserUpdateRequest }
+import cromwell.pipeline.datastorage.dto.{ User, UserWithCredentials }
+import cromwell.pipeline.model.wrapper.{ UserEmail, UserId }
+
+import scala.concurrent.Future
+
+class UserServiceTestImpl(users: Seq[UserWithCredentials], testMode: TestMode) extends UserService {
+
+  override def getUsersByEmail(emailPattern: String): Future[Seq[User]] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(users.map(user => User.fromUserWithCredentials(user)))
+    }
+
+  override def getUserWithCredentialsByEmail(email: UserEmail): Future[Option[UserWithCredentials]] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(users.headOption)
+    }
+
+  override def addUser(user: UserWithCredentials): Future[UserId] =
+    testMode match {
+      case WithException(exc) => Future.failed(exc)
+      case _                  => Future.successful(users.head.userId)
```

Comment:
If I create test implementation and call `addUser` like this
```scala
val user: UserWithCredentials = TestUserUtils.getDummyUserWithCredentials()
val userService = UserServiceTestImpl()()
userService.addUser(user)
```
Then I definitely don't expect this method to fail with `NoSuchElementException`

Reply:
Oops, fixed